### DISCUSSION TITLE
How to get metadata field searchable

### DISCUSSION THREAD

**Discussion by influence2006** [Security-Onion-Solutions/securityonion#5976] [Unsupported Versions]
SO 2.3.80
Cluster setup

I am trying to add the below picture to make it searchable. I have tried going to index patterns and them manually. This works temporarily but i guess after salt updates it goes back to not searchable. I have gone into ingest directory and edit the ossec to add one of the fields and do a so-elasticsearch-restart but the field is still not searchable. Can someone help me out, please? 

![image](https://user-images.githubusercontent.com/64173630/138355661-65d96677-c210-45c0-9f2d-41a319d62a3d.png)

**Comment by hacker0ni** [comment]
If you're sure about the Ingest node pipeline your log is using, you can copy it from `/opt/so/saltstack/default/salt/elasticsearch/files/ingest` to `/opt/so/saltstack/local/salt/elasticsearch/files/ingest/` then make the needed changes. After you save the file you need to restart ElasticSearch to apply changes. Since these files are managed by Salt, editing these settings/pipelines in the UI do not make a difference.

**Reply by influence2006** [reply]
I understand what you're saying. I am trying to figure out where to add these fields into the ossec file ( i am pretty sure I am looking at the right elasticsearch option) 

```
{
  "_index": "SecurityOnion:so-ossec-2021.10.22",
  "_type": "_doc",
  "_id": "gJcqqXwBo89f4umbwclJ",
  "_score": 1,
  "fields": {
    "program.source": [
      "clamav-0.103.3-5.el7.src.rpm"
    ],
    "program.architecture": [
      "x86_64"
    ],
    "program.vendor": [
      "Fedora Project"
    ],
```
the rename option failed on me miserably 
Do you have documentation I can read on the proper jinja format for these fields? 

trying to read through this to see how to get the proper reference to add those fields. 
https://www.elastic.co/guide/en/elasticsearch/reference/master/ingest.html


```
{ "set":            { "field": "program.architecture",               "value": "program_info"  } },
```

It doesnt fail when restarting but does not have the desired outcome.

**Reply by hacker0ni** [reply]
What you need to edit is the ingest pipelines, not the indexes themselves. Indexes generate these "aggregateable" or "queryable" values based on how they're ingested. If you include mapping these fields to custom fields and manipulate them with the Ingest pipeline, they should be queryable. From SOC, go to Kibana > Hamburger Menu > Stack Management > Ingest Node Pipelines and find your pipeline (should be `ossec`). Then go to manage, edit. Try playing around there and mapping your fields to different fields. You can also add a document to test, but that part's a little tricky. When you're done, you can use the `Show request` to grab the HTTP request along with the actual pipeline. Remove the first line (the PUT request) and copy the rest into `/opt/so/saltstack/local/salt/elasticsearch/files/ingest/ossec` . Restart elasticsearch to apply changes. Changes you make in Kibana Stack management won't apply. I'm planning on writing a 'creating your custom ingest pipeline' but that's a future project.
